internal/healthchecks/builtin.go

```diff
@@ -45,147 +47,38 @@ func AreObjectsHealthy(ctx context.Context, client client.Client, objects []clie
-			// Check if the daemonset is ready.
-			obj := &appsv1.DaemonSet{}
-			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, obj); err != nil {
-				gvkErrors = appendResourceError(gvkErrors, obj, err.Error())
-				continue
-			}
-			if obj.Status.NumberAvailable != obj.Status.DesiredNumberScheduled {
-				gvkErrors = appendResourceError(gvkErrors, obj, "DaemonSet is not ready")
-			}
-		case appsv1.SchemeGroupVersion.WithKind("ReplicaSet"):
-			// Check if the replicaset is ready.
-			obj := &appsv1.ReplicaSet{}
-			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, obj); err != nil {
-				gvkErrors = appendResourceError(gvkErrors, obj, err.Error())
-				continue
-			}
-			if obj.Status.AvailableReplicas != obj.Status.Replicas {
-				gvkErrors = appendResourceError(gvkErrors, obj, "ReplicaSet is not ready")
-			}
-		case corev1.SchemeGroupVersion.WithKind("Pod"):
-			// Check if the pod is running or succeeded.
-			obj := &corev1.Pod{}
-			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, obj); err != nil {
-				gvkErrors = appendResourceError(gvkErrors, obj, err.Error())
-				continue
-			}
-			if obj.Status.Phase != corev1.PodRunning && obj.Status.Phase != corev1.PodSucceeded {
-				gvkErrors = appendResourceError(gvkErrors, obj, "Pod is not Running or Succeeded")
-			}
-		case apiregistrationv1.SchemeGroupVersion.WithKind("APIService"):
 			// Check if the APIService is available.
-			obj := &apiregistrationv1.APIService{}
-			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, obj); err != nil {
-				gvkErrors = appendResourceError(gvkErrors, obj, err.Error())
-				continue
-			}
-			conditionExists := false
+			var isAvailable *apiregistrationv1.APIServiceCondition
 			for _, condition := range obj.Status.Conditions {
 				if condition.Type == apiregistrationv1.Available {
-					if condition.Status != "True" {
-						gvkErrors = appendResourceError(gvkErrors, obj, condition.Message)
-					}
-					conditionExists = true
-					break
-				}
-			}
-			if conditionExists {
-				continue
-			}
-			// If we are here we didn't find the "Available" condition, so we assume the APIService is non healthy.
-			gvkErrors = appendResourceError(gvkErrors, obj, "Available condition not found")
-		case apiextensionsv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"):
-			// Check if the CRD is established.
-			obj := &apiextensionsv1.CustomResourceDefinition{}
-			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, obj); err != nil {
-				gvkErrors = appendResourceError(gvkErrors, obj, err.Error())
-				continue
-			}
-			conditionExists := false
-			for _, condition := range obj.Status.Conditions {
-				if condition.Type == apiextensionsv1.Established {
-					if condition.Status != "True" {
-						gvkErrors = appendResourceError(gvkErrors, obj, condition.Message)
-					}
-					conditionExists = true
+					isAvailable = &condition
```

Comment:
As an alternative to @tmshort 's [recommendation](https://github.com/operator-framework/rukpak/pull/745/files#r1365973287) I _think_ you could also do something like:
```suggestion
					availCond := condition
					isAvailable = &availCond
```